gio/src/subclass/vfs.rs

```diff
@@ -0,0 +1,184 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::{prelude::*, subclass::prelude::*, translate::*, GString};
+
+use libc::c_char;
+
+use crate::{ffi, File, Vfs};
+
+// Support custom implementation of virtual functions defined in `gio::ffi::GVfsClass`.
+pub trait VfsImpl: ObjectImpl + ObjectSubclass<Type: IsA<Vfs>> {
+    fn is_active(&self) -> bool {
+        self.parent_is_active()
+    }
+
+    fn get_file_for_path(&self, path: impl AsRef<std::path::Path>) -> File {
```

Comment:
```suggestion
    fn get_file_for_path(&self, path: &std::path::Path) -> File {
```

would work equally well. `impl trait` in parameter position in the `FooImpl` traits is rather useless and only causes code bloat :)